workflow/controller/workflowpod_test.go

```diff
@@ -1246,10 +1246,16 @@ func Test_createSecretVolumesFromArtifactLocations_SSECUsed(t *testing.T) {
 	woc.operate(ctx)
 
 	mainCtr := woc.execWf.Spec.Templates[0].Container
-	pod, _ := woc.createWorkflowPod(ctx, wf.Name, []apiv1.Container{*mainCtr}, &wf.Spec.Templates[0], &createWorkflowPodOpts{})
-	assert.Contains(t, pod.Spec.Volumes, wantVolume)
-	assert.Len(t, pod.Spec.InitContainers, 1)
-	assert.Contains(t, pod.Spec.InitContainers[0].VolumeMounts, wantInitContainerVolumeMount)
+	for {
+		pod, _ := woc.createWorkflowPod(ctx, wf.Name, []apiv1.Container{*mainCtr}, &wf.Spec.Templates[0], &createWorkflowPodOpts{})
+		if pod != nil {
+			assert.Contains(t, pod.Spec.Volumes, wantVolume)
+			assert.Len(t, pod.Spec.InitContainers, 1)
+			assert.Contains(t, pod.Spec.InitContainers[0].VolumeMounts, wantInitContainerVolumeMount)
+			break
+		}
+	}
```

Comment:
I know this is just a test, but it seems kind of dangerous to have a infinite for loop that will only exit if createPod is successful. Maybe just try N times or use retry helper?

Reply:
Good idea. Updated